packages/amplify_datastore/ios/Classes/CognitoPlugin.swift

```diff
@@ -22,259 +22,140 @@ public class CognitoPlugin: AuthCategoryPlugin {
         self.nativeAuthPlugin = nativeAuthPlugin
     }
     
-
-    public func signIn(
-        withUrlUrl url: String,
-        callbackUrlScheme: String,
-        preferPrivateSession: NSNumber,
-        browserPackageName: String?
-    ) async -> ([String : String]?, FlutterError?) {
-        preconditionFailure("signIn is not supported")
-    }
-    
-    public func signOut(
-        withUrlUrl url: String,
-        callbackUrlScheme: String,
-        preferPrivateSession: NSNumber,
-        browserPackageName: String?
-    ) async -> FlutterError? {
-        preconditionFailure("signOut is not supported")
-    }
-    
     public let key: PluginKey = "awsCognitoAuthPlugin"
     
     public func configure(using configuration: Any?) throws {}
     
-    public func reset(onComplete: @escaping BasicClosure) {
-        onComplete()
+// TODO: Migrate to Async Swift v2
+//    public func reset(onComplete: @escaping BasicClosure) {
+//        onComplete()
+//    }
+    
+//    public func fetchAuthSession(
+//        options: AuthFetchSessionRequest.Options?,
+//        listener: ((AmplifyOperation<AuthFetchSessionRequest,
+//                        AuthSession,
+//                        AuthError>.OperationResult)
+//                  -> Void)?
+//    ) -> AuthFetchSessionOperation {
+//        let operation = NativeAuthFetchSessionOperation(
+//            categoryType: .auth,
+//            eventName: HubPayload.EventName.Auth.fetchSessionAPI,
+//            request: AuthFetchSessionRequest(options: options ?? AuthFetchSessionRequest.Options()),
+//            resultListener: listener
+//        )
+//        DispatchQueue.main.async {
+//            self.nativeAuthPlugin.fetchAuthSession { session in
+//                let result = NativeAWSAuthCognitoSession(from: session)
+//                operation.dispatch(result: .success(result))
+//            }
+//        }
+//        return operation
+//    }
+//
+//    public func getCurrentUser() -> AuthUser? {
+//        return currentUser
+//    }
```

Comment:
fetchAuthSession and getCurrentUser are commented out here, but implemented below with TODO statements. Do you need to keep this commented out code?

Reply:
Long term no. I left it here as a reference for the migration task. It should be removed then.